src/Pos/Block/Arbitrary.hs

```diff
@@ -378,20 +391,18 @@ newtype HeaderAndParams ssc = HAndP
 instance (Arbitrary (SscPayload ssc), SscHelpersClass ssc) =>
     Arbitrary (HeaderAndParams ssc) where
     arbitrary = do
-        consensus <- arbitrary :: Gen Bool
         -- This integer is used as a seed to randomly choose a slot down below
         seed <- arbitrary :: Gen Int
         (headers, leaders) <- first reverse . getHeaderList <$> arbitrary
         let num = length headers
         -- 'skip' is the random number of headers that should be skipped in the header
         -- chain. This ensures different parts of it are chosen each time.
         skip <- choose (0, num - 1)
-        let atMost3HeadersAndLeaders = take 3 $ drop skip headers
-            (prev, header, next) =
-                case atMost3HeadersAndLeaders of
-                    [h] -> (Nothing, h, Nothing)
-                    [h1, h2] -> (Just h1, h2, Nothing)
-                    (h1 : h2 : h3 : _) -> (Just h1, h2, Just h3)
+        let atMost2HeadersAndLeaders = take 2 $ drop skip headers
```

Comment:
`vhpConsensus` is fully checked in constructor.
`vhpNextHeader` wasn't used in code, so I removed it completely. There is no need to test something that is not used in actual code.